### ISSUE TITLE
Interacting with other instances is too hard

### ISSUE THREAD

**Issue by nazgum** [mastodon/mastodon#24068]
### Pitch

I do gamedev, but created my own instance so I start with no content to browse or ppl to find.  so i have to go to other instances, such as mastodon.gamedev.place to find ppl to follow and posts to interact with.  Now for every post i find there i may want to interact with, i need to copy that post url, go back to my instance paste it, then say star it, then go back to the other instance.  same with following ppl, bookmarking, replying or boosting content.  If im also interested in content from fosstodon I have to do the same thing there.

that is way too ackward.

I should be able to enter an instance url on my home instance, such as mastodon.gamedev.place or fosstodon.org, and be shown the local feed there of that instance, which then lets me follow ppl, boost posts, reply, etc. because i am logged in as me on my home instance.

Really, since a lot of instances are topical, a "Browse the Fediverse" btn could be added that brings up a list of instances, possibly sorted by topic or tags, and clicking those shows the local feed in your instance rather than going to it.  Your instance should be where everything happens, because that is where you are logged in.

### Motivation

If you run your own instance, are on a small instance, are interested in a different topic than the instance you joined, or are just new to mastodon, the usability of the Mastodon webapp seems really bad for following ppl on other instances and interacting with content on other instances.

Despite Mastodon being a federation of instances, you feel a bit locked into your own when u want to interact with content elsewhere.  That needs to be much easier, and if u are only logged in from your home instance, it needs to happen there - jumping back and forth every time pasting urls feels really bad.

**Comment by smiba** [comment]
I agree, as a workaround if you're using a desktop browser you can use https://github.com/Lartsch/FediAct

This allows you to interact on external instances without having to copy URLs and such

**Comment by chrisschm** [comment]
Agreed, but it may help to use an application to do so? I for my own use Fedilab (Android) to include the local timelines of other instances.

**Comment by supernovae** [comment]
As I was sailing around on peaceful waters and had more time to think on stuff like this it occurred to me we need to formalize something simple and then form our improvements around it.

**Are we "Mastodon the social network" where people expect completeness across the social network**

or

**Are we "mastodon the software" where completeness of local community is all that matters?**

In my brain, I am firmly in "mastodon the social network" since that is how i felt it was framed.  I expect well federated replies, likes and boosts because those shape "the social network"

Lots of admins believe in "mastodon the software" where the social network is the local community.

I've considered asking for options so admins could opt into behaviors (and define this on their about page) such as "well federated" or "loosely federated" as I'm perfectly fine with there being niche islands of people who don't care about posts, likes and boosts from elsewhere but still want to participate as loosely coupled.

**Comment by trwnh** [comment]
at the risk of repeating various discussions whose issues i am too lazy to go track down every single one:

- on the technical layer, the "network" is the World Wide Web
- on the social layer, for a "mastodon network" to exist, you would need either a relay service to distribute posts to all participants in the relay, or you would need an aggregator service that tracks all your posts. the "network" at that point is the central relay/aggregator, operated by mastodon ggmbh and subject to its own policies and moderation. 

i don't think you can meaningfully have a "mastodon network" outside of that framework. replication and distribution can't happen without cooperation and shared trust. on a universal level, consensus and consistency are traded away for availability and partition tolerance, so the "network" doesn't go down if any part of it goes down, and it continues to operate even if some messages are missing (per CAP theorem).

so it's pretty clearly "mastodon the software", which lets you run your own network that talks to other networks. changing this would require rearchitecting the software to work almost completely differently. not to say this can't be done, but that this requires a great deal of effort.

specific usability concerns like OP discusses, with browsing to a domain's website and wanting to interact from another domain, can be addressed with some form of cross-domain identity verification or authorization: indieauth, remote interaction dialogs, and so on. but these require at minimum that you provide your identity to every website you visit, either by profile URL or by username@domain. the former would check the HTML response for endpoints exposed via link relations, such as `<link rel="authorization_endpoint" href="..."/>`. see the later parts of #4800 for more. the latter would perform a webfinger resolution and get some interaction URI to load the object locally. see the comments on #23970 for a proposal on formalizing `https://www.w3.org/ns/activitystreams#proxyUrl` for this purpose.

**Comment by KaKi87** [comment]
Hello,

I've just released [Fediverse redirector](https://git.kaki87.net/KaKi87/userscripts/src/branch/master/fediverseRedirector/README.md), a userscript that, among other things, automatically redirects all Mastodon profile/post links to your favourite instance.

May it make your interactions easier. :)

**Comment by Toastyyy3** [comment]
Is there any new discussion on this? I'm on the verge of switching to the Fediverse, however this is one major point which holds me back. I know the federation to that extent is also considered a feature but as @supernovae already mentioned, it'd be amazing to have some kind of opt-in or opt-out just for it to be appealing for both kind of users.
